src/engine/AbsoluteValueConstraint.cpp

```diff
@@ -0,0 +1,454 @@
+/*********************                                                        */
+/*! \file ReluConstraint.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Shiran Aziz, Guy Katz
+ ** This file is part of the Marabou project.
+ ** Copyright (c) 2017-2019 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved. See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** [[ Add lengthier description here ]]
+ **/
+
+#include "AbsoluteValueConstraint.h"
+#include "ConstraintBoundTightener.h"
+#include "Debug.h"
+#include "FloatUtils.h"
+#include "ITableau.h"
+#include "MStringf.h"
+#include "MarabouError.h"
+#include "PiecewiseLinearCaseSplit.h"
+#include "Statistics.h"
+
+AbsoluteValueConstraint::AbsoluteValueConstraint( unsigned b, unsigned f )
+    : _b( b )
+    , _f( f )
+    , _haveEliminatedVariables( false )
+{
+    setPhaseStatus( PhaseStatus::PHASE_NOT_FIXED );
+}
+
+PiecewiseLinearConstraint *AbsoluteValueConstraint::duplicateConstraint() const
+{
+    AbsoluteValueConstraint *clone = new AbsoluteValueConstraint( _b, _f );
+    *clone = *this;
+    return clone;
+}
+
+void AbsoluteValueConstraint::restoreState( const PiecewiseLinearConstraint *state )
+{
+    const AbsoluteValueConstraint *abs = dynamic_cast<const AbsoluteValueConstraint *>( state );
+    *this = *abs;
+}
+
+void AbsoluteValueConstraint::registerAsWatcher( ITableau *tableau )
+{
+    tableau->registerToWatchVariable( this, _b );
+    tableau->registerToWatchVariable( this, _f );
+}
+
+void AbsoluteValueConstraint::unregisterAsWatcher( ITableau *tableau )
+{
+    tableau->unregisterToWatchVariable( this, _b );
+    tableau->unregisterToWatchVariable( this, _f );
+}
+
+void AbsoluteValueConstraint::notifyVariableValue( unsigned variable, double value )
+{
+    _assignment[variable] = value;
+}
+
+/*
+  The variable watcher notification callbacks, about a change in a variable's value or bounds.
+  suppose A < x_b < B, C < x_f < D
+  if variable == x_b then:
+    A < 0 & B < 0 then: max{|B|, C} < x_f < min{|A|, D}
+    A < 0 & B > 0 then: max{0, C} < x_f < min{max{|A|, B}, D}
+    A > 0 & B < 0 then: ------
```

Comment:
Spell out that  -----  means UNSAT/Infeasable/Conflict

Reply:
The code comment was stale, removing it.